x-pack/plugin/esql/qa/server/multi-clusters/build.gradle

```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import org.elasticsearch.gradle.Version
+import org.elasticsearch.gradle.internal.info.BuildParams
+import org.elasticsearch.gradle.testclusters.StandaloneRestIntegTestTask
+
+apply plugin: 'elasticsearch.internal-java-rest-test'
+apply plugin: 'elasticsearch.bwc-test'
+
+dependencies {
+  javaRestTestImplementation project(xpackModule('esql:qa:testFixtures'))
+  javaRestTestImplementation project(xpackModule('esql:qa:server'))
+}
+
+def ccsSupportedVersion = bwcVersion -> {
```

Comment:
So as I read this, only the 9.0 release will be compatible with the last 8.x, but not 9.1+. Is that correct?

Reply:
Both 9.0 and 9.1 should work with the latest 8.x. This module is a one-way BWC with the local cluster on the latest version and the remote cluster on the BWC version. I will consider adding another module where the local cluster is on the BWC version. Here I just followed (copied from an existing CCS QA module) the CCS compatibility matrix: https://www.elastic.co/guide/en/elasticsearch/reference/current/modules-cross-cluster-search.html#ccs-supported-configurations.

Reply:
Ok, that doesn't align with how I'm reading the logic then.

```
currentVersion.minor == 0 || (currentVersion.major == bwcVersion.major && currentVersion.minor - bwcVersion.minor <= 1)
```

If the current version is `9.1` then no `8.x` version would be supported.

Reply:
Yes, the existing logic in the multi-cluster-search appears to be incorrect. I copied this from there. I've relaxed this to test with all wired compatible versions instead of the CCS matrix. I think we might add a `withCCSCompatible` option to our build system.